traffic_ops/traffic_ops_golang/deliveryservice/keys.go

```diff
@@ -210,9 +259,75 @@ func getSSLKeysByXMLIDHelper(xmlID string, inf *api.APIInfo, w http.ResponseWrit
 			return
 		}
 	}
+
 	api.WriteResp(w, r, keyObj)
 }
 
+// GetSSLKeysByXMLID fetches the deliveryservice ssl keys by the specified xmlID. V14 includes expiration date.
+func GetSSLKeysByXMLIDV15(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"xmlid"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	if inf.Config.RiakEnabled == false {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusServiceUnavailable, errors.New("the Riak service is unavailable"), errors.New("getting SSL keys from Riak by xml id: Riak is not configured"))
```

Comment:
Same as above RE returning opaque server-side errors to the client

Reply:
done